network/overlay/ensure.go

```diff
@@ -40,14 +55,20 @@ func (netm manager) Ensure(ctx context.Context, network types.Network) error {
 		return errors.Wrapf(err, "fail to get netlink handler of netns")
 	}
 	defer nlh.Delete()
+	err = nlh.SetSocketTimeout(NetlinkSocketsTimeout)
+	if err != nil {
+		return errors.Wrapf(err, "fail to configure timeout on netlink socket")
+	}
```

Comment:
Why do we need to set this timeout? What is it for?

Reply:
It'a what libnetwork is doing, to prevent having infinite timeout sockets according to their documentation, just like http client timeout